src/Dictionaries/LibraryDictionarySource.cpp

```diff
@@ -193,15 +192,20 @@ BlockInputStreamPtr LibraryDictionarySource::loadAll()
         columns.data[i] = a.name.c_str();
         ++i;
     }
+
     void * data_ptr = nullptr;
 
     /// Get function pointer before dataNew call because library->get may throw.
     auto func_load_all
         = library->get<void * (*)(decltype(data_ptr), decltype(&settings->strings), decltype(&columns))>("ClickHouseDictionary_v3_loadAll");
+
     data_ptr = library->get<decltype(data_ptr) (*)(decltype(lib_data))>("ClickHouseDictionary_v3_dataNew")(lib_data);
+    SCOPE_EXIT(library->get<void (*)(decltype(lib_data), decltype(data_ptr))>("ClickHouseDictionary_v3_dataDelete")(lib_data, data_ptr));
```

Comment:
@alexey-milovidov, @alesapin  I assume if dataToBlock function throws exception before we does not call deleter for data that we get from shared library.